src/main/kotlin/com/openlattice/chronicle/services/timeusediary/TimeUseDiaryService.kt

```diff
@@ -0,0 +1,157 @@
+        private val objectMapper = ObjectMapper()
+        private val cal = Calendar.getInstance()
+    }
+
+    override fun submitTimeUseDiary(
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        responses: List<TimeUseDiaryResponse>
+    ): UUID {
+        val tudId = UUID.randomUUID()
+        try {
+            val (flavor, hds) = storageResolver.resolve(studyId)
+            hds.connection.use { conn ->
+                executeInsertTimeUseDiarySql(
+                    conn,
+                    tudId,
+                    organizationId,
+                    studyId,
+                    participantId,
+                    responses
+                )
+            }
+        } catch(e: Exception) {
+            logger.error("hds error: $e")
+        }
+        return tudId
+    }
+
+    override fun getSubmissionByDate(
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        startDate: OffsetDateTime,
+        endDate: OffsetDateTime
+    ): Map<LocalDate, Set<UUID>> {
+        val submissionsByDate = mutableMapOf<LocalDate,MutableSet<UUID>>()
+        // Use Calendar to convert response to user's timezone
+        cal.timeZone = TimeZone.getTimeZone("GMT${startDate.offset.id}")
+        try {
+            val (flavor, hds) = storageResolver.resolve(studyId)
+            val result = hds.connection.use { connection ->
+                executeGetSubmissionByDateSql(
+                    connection,
+                    organizationId,
+                    studyId,
+                    participantId,
+                    startDate,
+                    endDate
+                )
+            }
+            while (result.next()) {
+                val currentDate = result.getDate(SUBMISSION_DATE.name, cal).toLocalDate()
+                val currentUUID = UUID.fromString(result.getString(TUD_ID.name))
+                if (submissionsByDate[currentDate].isNullOrEmpty()) {
+                    submissionsByDate[currentDate] = mutableSetOf(currentUUID)
+                } else {
+                    submissionsByDate[currentDate]!!.add(currentUUID)
+                }
+            }
+        } catch (e: Exception) {
+            logger.error("hds error: $e")
+        }
+        return submissionsByDate
+    }
+
+    override fun downloadTimeUseDiaryData(
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        type: TimeUseDiaryDownloadDataType,
+        submissionsIds: Set<UUID>
+    ) {
+        TODO("Not yet implemented")
+    }
+
+    /* -------- SQL helpers -------- */
+
+    private fun executeInsertTimeUseDiarySql(
+        connection: Connection,
+        tudId: UUID,
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        responses: List<TimeUseDiaryResponse>
+    ) {
+
+       val preparedStatement = connection.prepareStatement("""
```

Comment:
Please refactor the SQL into a Dokka documented comment with the parameters annotated.